core/include/scipp/core/element/to_unit.h

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+
+#include "scipp/common/overloaded.h"
+#include "scipp/core/element/arg_list.h"
+#include "scipp/core/time_point.h"
+#include "scipp/core/transform_common.h"
+
+namespace scipp::core::element {
+
+namespace {
+template <class T>
+constexpr auto round =
+    [](const auto x) { return static_cast<T>(x < 0 ? x - 0.5 : x + 0.5); };
```

Comment:
Is this the same as `std::round`?

Reply:
No, unfortunately, at least I was not successful: There is `std::lround` and `std::llround`, but nothing with a definite bit size (32 or 64). I had tried using `lround` when for `int32_t`, but when out of domain it would result in some integer that was about int32_max/2 (not exactly).

Reply:
Ok.
Maybe put in a check if `T` is integral so that it is a noop in that case instead of converting to float and back.